docs/src/tutorials/testing.md

```diff
@@ -1,243 +1,505 @@
+	# Create 2-node-network (l1)----(l2)
+    l1, l2 = node_factory.line_graph(2)
+    l2.fundwallet(sats=2*10**6)
+
+	# Register a new Greenlight client.
+    c = clients.new()
+    c.register(configure=True)
+    gl1 = c.node()
+
+	# Start signer and connect to (l2)
+    s = c.signer().run_in_thread()
+    gl1.connect_peer(l2.info['id'], f'127.0.0.1:{l2.daemon.port}')
+
+	# Fund a channel (l2)----(gl).
+	# This results in the following network
+	# (l1)----(l2)----(gl)
+    l2.rpc.fundchannel(c.node_id.hex(), 'all')
+	# Generate a block to synchronize and proceed
+	# with the channel funding.
+    bitcoind.generate_block(1, wait_for_mempool=1)
+	# Wait for the channel to confirm.
+    wait_for(lambda:
+        gl1.list_peers().peers[0].channels[0].state == 'CHANNELD_NORMAL'
+    )
 ```
 
-This will poll `gl1` for its channel states and return as soon as the
-state indicates that the channel is confirmed and fully functional.
+Before we create and pay the invoice we also need to wait for the 
+gossip to reach every node. Otherwise the invoice will be lacking
+route hints as it assumes its only channel to be a dead end.
+Alternatively we could also create the invoice directly and wait for 
+l1 to have a full view of our small network but lets go with the 
+first option this time. We again use the `wait_for` function. The
+successor function checks that we see 4 channel entries in our view
+of the network as both channels are bidirectional.
 
-Finally we can create an invoice from the Greenlight node `gl1` and
-pay it using `l1`, which should result in a path `l1 -> l2 -> gl1`
-being used, and the invoice will be paid after this:
+```py linenums="1" hl_lines="30-38"  title="my-first-test.py"
+from gltesting.fixtures import *
+from pyln.testing.utils import wait_for
+
+def test_invoice_payment(node_factory, clients, bitcoind):
+	# Create 2-node-network (l1)----(l2)
+    l1, l2 = node_factory.line_graph(2)
+    l2.fundwallet(sats=2*10**6)
+
+	# Register a new Greenlight client.
+    c = clients.new()
+    c.register(configure=True)
+    gl1 = c.node()
+
+	# Start signer and connect to (l2)
+    s = c.signer().run_in_thread()
+    gl1.connect_peer(l2.info['id'], f'127.0.0.1:{l2.daemon.port}')
+
+	# Fund a channel (l2)----(gl).
+	# This results in the following network
+	# (l1)----(l2)----(gl)
+    l2.rpc.fundchannel(c.node_id.hex(), 'all')
+	# Generate a block to synchronize and proceed
+	# with the channel funding.
+    bitcoind.generate_block(1, wait_for_mempool=1)
+	# Wait for the channel to confirm.
+    wait_for(lambda:
+        gl1.list_peers().peers[0].channels[0].state == 'CHANNELD_NORMAL'
+    )
+
+	# Wait for all channels to appear in our view of the network. We 
+    # don't even have to wait for our channel to appear in l1s 
+	# gossip: We can give a hint as soon as we know that our channel 
+	# is not a dead end. We wait for 4 entries in our gossmap as both 
+    # channels are bidirectional.
+    bitcoind.generate_block(5)
```

Comment:
You can also just call `line_graph` with `wait_for_announce=True` and the `node_factory` will take care of confirming and announcing the channels.